src/factories/BaseFactory.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.17;
+
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+
+/**
+ * @title BaseFactory This is the base contract for all Oval factories. It manages the maxTraversal and default
+ * unlockers used by all Oval factories.
+ * @dev Derived contracts should implement a create method with the parameters needed to instantiate that flavor of
+ * Oval.
+ */
+contract BaseFactory is Ownable {
+    uint256 public immutable MAX_TRAVERSAL;
+    address[] public defaultUnlockers;
+
+    event DefaultUnlockersSet(address[] defaultUnlockers);
+    event OvalDeployed(
+        address indexed deployer,
+        address indexed oval,
+        uint256 indexed lockWindow,
+        uint256 maxTraversal,
+        address owner,
+        address[] unlockers
+    );
+
+    constructor(uint256 maxTraversal, address[] memory _defaultUnlockers) {
+        MAX_TRAVERSAL = maxTraversal;
```

Comment:
I don't really see max traversal changing that often for a particular oracle type. If it were to change, I think it would be acceptable to just move to a new factory.